contracts/governance/ProposalValidator.sol

```diff
@@ -3,198 +3,253 @@ pragma solidity 0.7.6;
+    if (proposal.proposalType != IKyberGovernance.ProposalType.Binary) return false;
+    return (proposal.voteCounts[YES_INDEX].mul(ONE_HUNDRED_WITH_PRECISION).div(proposal.maxVotingPower) >
+      proposal.voteCounts[NO_INDEX].mul(ONE_HUNDRED_WITH_PRECISION).div(proposal.maxVotingPower).add(
+        VOTE_DIFFERENTIAL
+      ));
   }
 
   /**
-   * @dev Check whether a proposal has enough extra FOR-votes than AGAINST-votes
-   * FOR VOTES - AGAINST VOTES > VOTE_DIFFERENTIAL * voting supply
+   * @dev Fetch the winning option of a generic proposal
    * @param governance Governance Contract
    * @param proposalId Id of the proposal to verify
-   * @return true if enough For-Votes
+   * @return option index with the most votes and sufficient quorum, 0 otherwise
    **/
-  function isVoteDifferentialValid(IKyberGovernance governance, uint256 proposalId)
-    public
-    override
+  function getGenericProposalWinningOption(IKyberGovernance governance, uint256 proposalId)
+    external
     view
-    returns (bool)
+    override
+    returns (uint256 winningOption)
   {
-    IKyberGovernance.ProposalWithoutVote memory proposal = governance.getProposalById(
-      proposalId
-    );
-    uint256 votingSupply = IGovernanceStrategy(proposal.strategy).getTotalVotingSupplyAt(
-      proposal.startBlock
-    );
+    IKyberGovernance.ProposalWithoutVote memory proposal = governance.getProposalById(proposalId);
+    if (proposal.proposalType != IKyberGovernance.ProposalType.Generic) return 0;
 
-    return (proposal.forVotes.mul(ONE_HUNDRED_WITH_PRECISION).div(votingSupply) >
-      proposal.againstVotes.mul(ONE_HUNDRED_WITH_PRECISION).div(votingSupply).add(
-        VOTE_DIFFERENTIAL
-      ));
+    uint256 winningVoteCount;
+    (winningOption, winningVoteCount) = getWinningOptionData(proposal.voteCounts);
+    if (winningOption == 0) return 0;
+    if (!isMinimumQuorumReached(winningVoteCount, proposal.maxVotingPower)) return 0;
+
+    // as we already limit value for c & t, no need to check for overflow here
+    uint256 votedPercentage = proposal.totalVotes.mul(PRECISION).div(proposal.maxVotingPower);
+    uint256 x = T_IN_PRECISION.mul(votedPercentage).div(PRECISION);
+    if (x <= C_IN_PRECISION) {
+      // threshold is not negative, need to compare with voted count
+      uint256 y = C_IN_PRECISION.sub(x);
+      // (most voted option count / total votes) is below threshold, no winining option
+      if (winningVoteCount.mul(PRECISION) < y.mul(proposal.totalVotes)) {
+        return 0;
+      }
+    }
+  }
+
+  function getWinningOptionData(uint256[] voteCounts)
+    internal
+    pure
+    returns (uint256 winningOption, uint256 winningVoteCount)
+  {
+    uint256 maxVotedCount;
+    for (uint256 i = 0; i < voteCounts.length; i++) {
+      if (voteCounts[i] > maxVotedCount) {
+        winningOption = i + 1;
+        maxVotedCount = voteCounts[i];
+        winningVoteCount = maxVotedCount;
+      } else if (voteCounts[i] == maxVotedCount) {
```

Comment:
seems like an issue from previous KyberDAO here, if `voteCounts[0]` and `voteCounts[1]` have the same value (but not the highest), then it will have 0 winning option.
Need to get the highest vote first, then check if there are more than 1 options with the same highest votes